src/miniscript/satisfy.rs

```diff
@@ -401,15 +563,19 @@ impl Ord for Witness {
 
 impl Witness {
     /// Turn a signature into (part of) a satisfaction
-    fn signature<Pk: ToPublicKey, S: Satisfier<Pk>>(sat: S, pk: &Pk) -> Self {
-        match sat.lookup_sig(pk) {
-            Some((sig, hashtype)) => {
-                let mut ret = sig.serialize_der().to_vec();
-                ret.push(hashtype.as_u32() as u8);
-                Witness::Stack(vec![ret])
+    fn signature<Pk: ToPublicKey, S: Satisfier<Pk>, Ctx: ScriptContext>(sat: S, pk: &Pk) -> Self {
+        if Ctx::is_tap() {
```

Comment:
With more post-TapScript contexts in the future we will get a bug in this code and compiler will not warn/error. So I propose to use `match` here and not `if`

Reply:
Agreed.

Reply:
I think the last commit should make this cleaner. Let me know if you want me to reorder the commits.